controllers/s3utils.go

```diff
@@ -80,52 +81,79 @@ import (
 // }
 // }
 
-type s3Connection struct {
-	session    *session.Session
-	client     *s3.S3
-	uploader   *s3manager.Uploader
-	downloader *s3manager.Downloader
-	endpoint   string
-	callerTag  string
+// ObjectStoreGetter interface is exported because AVR test,
```

Comment:
This comment is incorrect.  We should not expose an interface because of a test.  It is the other way around.  We expose an interface to be used by clients, be it a real client or a test client.  The original code didn't need this interface.  I am not saying this is wrong, but I am saying and putting the emphasis on exposing an interface in order to abstract the implementation and to provide separation.

Reply:
I agree with the sentiment of your comment. At the time I wrote that comment, I didn't realize that Ginkgo by default suggests that tests be in a different _test package (that it doesn't mandate it is a different matter). We wouldn't expose the interface if the tests were in the same package.  At that time, the interface was only exported because AVR test needed, but I didn't realize that soon VRG too would use it if it also followed the Ginkgo suggested pattern of a separate test package!  In any case, I will remove that emphasis.